owslib/csw.py

```diff
@@ -608,12 +616,17 @@ def _invoke(self):
                 if caller == 'getrecords2': caller = 'getrecords'
                 try:
                     op = self.get_operation_by_name(caller)
-                    post_verbs = filter(lambda x: x.get('type').lower() == 'post', op.methods)
+                    post_verbs = [x for x in op.methods if x.get('type').lower() == 'post']
                     if len(post_verbs) > 1:
                         # Filter by constraints.  We must match a PostEncoding of "XML"
-                        try:
-                            xml_post_url = next(x for x in filter(list, ([pv.get('url') for const in pv.get('constraints') if const.name.lower() == "postencoding" and 'xml' in map(lambda x: x.lower(), const.values)] for pv in post_verbs)))[0]
```

Comment:
This is one place where I did rewrite the code a fair bit. The enormous single line is very hard to follow, so I expanded it to a few lines of regular code - please check the logic carefully.